samples/kvssink_intermittent_sample.cpp

```diff
@@ -0,0 +1,293 @@
+#include <thread>
+#include <mutex>
+#include <chrono>
+#include <condition_variable>
+
+#include <gst/gst.h>
+#include <glib.h>
+
+#include "gstreamer/gstkvssink.h"
+
+using namespace com::amazonaws::kinesis::video;
+using namespace log4cplus;
+
+/* modify these values to change start/stop interval */
+#define KVS_INTERMITTENT_PLAYING_INTERVAL_SECONDS 20
+#define KVS_INTERMITTENT_PAUSED_INTERVAL_SECONDS 40
+
+
+LOGGER_TAG("com.amazonaws.kinesis.video.gstreamer");
+
+GMainLoop *main_loop = g_main_loop_new (NULL, FALSE);
+std::atomic<bool> terminated(FALSE);
+std::condition_variable cv;
+
+typedef enum _StreamSource {
+    TEST_SOURCE,
+    DEVICE_SOURCE
+} StreamSource;
+
+typedef struct _CustomData {
+    _CustomData() :
+            main_loop(NULL),
+            pipeline(NULL) {}
+
+    GMainLoop *main_loop;
+    GstElement *pipeline;
+} CustomData;
+
+void sigint_handler(int sigint){
+    LOG_DEBUG("SIGINT received.  Exiting...");
+    terminated = TRUE;
+    cv.notify_all();
+    if(main_loop != NULL){
+        g_main_loop_quit(main_loop);
+    }
+}
+
+static gboolean
+bus_call (GstBus *bus, GstMessage *msg, gpointer data)
+{
+    GMainLoop *loop = (GMainLoop *) ((CustomData *)data)->main_loop;
+    GstElement *pipeline = (GstElement *) ((CustomData *)data)->pipeline;
+
+    switch (GST_MESSAGE_TYPE (msg)) {
+        case GST_MESSAGE_EOS: {
+            LOG_DEBUG("[KVS sample] Received EOS message");
+            cv.notify_all();
+            break;
+        }
+
+        case GST_MESSAGE_ERROR: {
+            gchar  *debug;
+            GError *error;
+
+            gst_message_parse_error (msg, &error, &debug);
+            g_free (debug);
+
+            LOG_ERROR("[KVS sample] GStreamer error: " << error->message);
+            g_error_free (error);
+
+            g_main_loop_quit (loop);
+            break;
+        }
+
+        default: {
+            break;
+        }
+    }
+
+    return TRUE;
+}
+
+void determine_aws_credentials(GstElement *kvssink, char* streamName) {
+    char const *iot_credential_endpoint;
+    char const *cert_path;
+    char const *private_key_path;
+    char const *role_alias;
+    char const *ca_cert_path;
+    char const *credential_path;
+    if (nullptr != (iot_credential_endpoint = getenv("IOT_GET_CREDENTIAL_ENDPOINT")) &&
```

Comment:
Use capital GETENV (macro) from PIC.

We also have IS_NULL_OR_EMPTY_STRING macro as well.